waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -525,16 +525,23 @@
                        :service-id service-id}))))
   (let [rs-spec (replicaset-spec-builder-fn scheduler service-id service-description)
         request-namespace (k8s-object->namespace rs-spec)
-        request-url (str api-server-url "/apis/" replicaset-api-version "/namespaces/"
-                         request-namespace "/replicasets")
+        request-url (str api-server-url "/apis/" replicaset-api-version "/namespaces/" request-namespace "/replicasets")
         response-json (api-request request-url scheduler
                                    :body (utils/clj->json rs-spec)
                                    :request-method :post)
-        service (some-> response-json replicaset->Service)]
+        {:keys [k8s/replicaset-uid] :as service} (some-> response-json replicaset->Service)]
     (when-not service
       (throw (ex-info "failed to create service"
                       {:service-description service-description
                        :service-id service-id})))
+    (let [{:strs [min-instances]} service-description]
+      (when (> min-instances 1)
+        (let [pdb-spec (pdb-spec-builder-fn pdb-api-version rs-spec replicaset-uid)
+              request-url (str api-server-url "/apis/" pdb-api-version "/namespaces/" request-namespace "/poddisruptionbudgets")]
+          (log/info "creating pod disruption budget with min available instance of one for" service-id)
+          (api-request request-url scheduler
```

Comment:
Do we want a try/catch around this? Right now the function will fail if we fail to create the PDB. Is that expected and/or desired?

Reply:
I don't think we want to fail to return the `service` just because the PDB creation failed.